### ISSUE TITLE
Content channel disappears after UPDATE

### ISSUE THREAD

**Issue by ivanistheone** [learningequality/kolibri#3246]
WHERE: on http://35.229.41.226/  (creds `devowner:admin123`) which runs Kolibri 0.7.2b2.dev+git-1-gdd27d31

### Observed behavior

Trying to UPDATE a channel makes it disappear 

### Expected behavior

Channel remains and contains updated content.

### User-facing consequences

Users cannot update content channels. Will have to delete then re-import.

### Errors and logs

Nothing interesting in logs:

```
INFO:kolibri.content.management.commands.importchannel:Downloading data for channel id 75e2205cb9c449c99a12e558520e5eb1
^M  0%|          | 0/55296 [00:00<?, ?it/s]^[[37mINFO     Channel 75e2205cb9c449c99a12e558520e5eb1 already exists in database, cleaning up ContentNodes^[[0m
INFO:kolibri.content.utils.channel_import:Channel 75e2205cb9c449c99a12e558520e5eb1 already exists in database, cleaning up ContentNodes
INFO:kolibri.content.utils.channel_import:Importing ContentTag data
INFO:kolibri.content.utils.channel_import:Importing ContentNode_has_prerequisite data
INFO:kolibri.content.utils.channel_import:Importing ContentNode_related data
INFO:kolibri.content.utils.channel_import:Importing ContentNode_tags data
INFO:kolibri.content.utils.channel_import:Importing ContentNode data
INFO:kolibri.content.utils.channel_import:Importing Language data
INFO:kolibri.content.utils.channel_import:Importing File data
INFO:kolibri.content.utils.channel_import:Importing LocalFile data
INFO:kolibri.content.utils.channel_import:Importing AssessmentMetaData data
INFO:kolibri.content.utils.channel_import:Importing ChannelMetadata data
INFO:kolibri.content.utils.annotation:Setting availability of File objects based on LocalFile availability
INFO:kolibri.content.utils.annotation:Setting availability of non-topic ContentNode objects based on File availability
INFO:kolibri.content.utils.annotation:Setting availability of ContentNode objects with children for 3 levels
INFO:kolibri.content.utils.annotation:Setting availability of ContentNode objects with children for level 3
INFO:kolibri.content.utils.annotation:Setting availability of ContentNode objects with children for level 2
INFO:kolibri.content.utils.annotation:Setting availability of ContentNode objects with children for level 1
```

### Steps to reproduce

initial state
<img width="1348" alt="1 initial state" src="https://user-images.githubusercontent.com/163966/36608386-abeda52c-1897-11e8-9dcc-1a83259a7e05.png">

select channel to import
<img width="939" alt="2 select channel for import" src="https://user-images.githubusercontent.com/163966/36608398-b2fe0938-1897-11e8-8f77-7964482b6761.png">

prompted for update
<img width="1353" alt="3 prompted to update" src="https://user-images.githubusercontent.com/163966/36608406-bc7133be-1897-11e8-8fa6-15b42355042c.png">

after completing update channel is no longer there
<img width="940" alt="4 after completing update channel is no longer there" src="https://user-images.githubusercontent.com/163966/36608412-c10c6b1e-1897-11e8-945e-878b398dfc5e.png">

**Comment by jonboiser** [comment]
@ivanistheone sacre bleu! it looks like the channel gets switched to unavailable after the update. see: http://35.229.41.226/api/channel/

@lyw07 Didn't we see something like this happen before?

**Comment by lyw07** [comment]
@jonboiser I'm sorry I can't recall if this happened before or the reason of this issue.
@ivanistheone I tried to update another channel on http://35.229.41.226/, and it didn't disappear. 
Do you know what is changed in the channel for the new version? Thank you!

**Comment by jonboiser** [comment]
I don't know what the exact logic is for annotating a Channel as 'available', but it basically means the Channel has at least one Content-File in the File system.

Since this testing channel looks like it has just one file, if the 'update' is to replace that one file with something else completely different, I can imagine that, post-update, the Channel will be annotated as 'unavailable'.

**Comment by ivanistheone** [comment]
@lyw07  Okay, just uploaded, published, and imported a new version of the channel "Sample Ricecooker Channel" and it did not disappear after import. Pheeew. Bug severity--.

This issue could be related to the fact that the "Testing math formatting" channel has only a single exercise node, and no other content nodes.

**Comment by lyw07** [comment]
@jonboiser I think you are right. I tried with replace the existing file with a new file from the Kolibri Studio side, and the channel disappeared after updating.

**Comment by ivanistheone** [comment]
I ran into the disappearing channel bug again today (Updating a channel with a single exercise node in removes it from Kolibri installed channels).

It seems the "already on your device" logic used during content update is not working 100%, at least not for exercise nodes.

I was able to "break" a channel during UPDATE. I had modified an exercise (which changes the perseus json file for the question), but the import wizard showed the channel as "No changes"

After this partial-Update, going to the exercise under `/learn` shows an error in the browser, presumably because it's trying to load a perseus.json file it hasn't synced.

<img width="859" alt="screen shot 2018-03-21 at 3 48 20 pm" src="https://user-images.githubusercontent.com/163966/37733713-5ed2a6c4-2d1f-11e8-956f-04fc1bc68323.png">


I guess what's going on is:
  - the node is marked as available since the exercise + associated perseus files was previous imported
  - the first part of the channel update process updates the DB tables
  - by not doing the second part of the channel update (importing the files), we end up with a broken state since `assessment_item_ids` column in `content_assessmentmetadata` points to file that doesn't exist.
